enhancements/installer/telemeter-metrics.md

```diff
@@ -0,0 +1,430 @@
+ > 3. How to authenticate using pull secret.
+ 
+## Summary
+Telemetry collects metrics from running clusters, but no metrics are obtained from the installer binary.
+This enhancement proposes that the installer would push metrics related to user workflows, installation statistics, and feature usage. The installer would be instrumented to collect Prometheus label key-value pairs, convert them to metric samples and push those metrics to an [aggregation pushgateway][weaveworks-aggregation-pushgateway].
+This enhancement focuses on the design of the metrics to be pushed to Prometheus.
+
+## Motivation
+
+Obtaining metrics from the installer would allow engineers and product managers to understand different user workflows, environments, experiences, and difficulties, as well as an indication of feature usage.  By pushing these metrics to the Telemeter, OpenShift members would be able to create a variety of queries related to these aspects.
+
+### Goals
+
+1. Design meaningful metrics to collect user experience and behavior with the installer
+
+### Non-Goals
+
+1. Collected metrics will not allow analysis of individual installer invocations and will not display detailed data such as log or error messages. We won't be able to answer "how long did an individual installation take?" but rather 95% of installs finish within, say, 40 minutes.
+2. Installer instrumentation/internal implementation will be discussed in a separate design/enhancement
+3. Pushgateway implementation will be discussed in a separate design/enhancement
+
+## Proposal
+
+### User Stories
+
+#### Telemetry Analyst
+Some examples:
+
+As a telemetry analyst I would like to create queries which analyze aggregate installation data to determine how long the installation process takes in total and broken down by different stages so that I can better understand user experience, expose any abnormalities, and compare different versions of the installer.
+
+As a telemetry analyst I would like to create queries which show aggregate counts of failure by stage of installation so that we can better understand where users are experiencing problems. (This will not provide  specific error messages/logs).
+
+As a telemetry analyst I would like to see how often users modify ignition configs or manifests so that I can better understand how users are interacting with the installer. [(see open question #3)](open-questions)
+
+As a telemetry analyst I would like to see how often the installer is being run on certain operating systems so I can better understand our user environments and needs.
+
+### Implementation Details/Notes/Constraints
+The installer does not fit a typical Prometheus use case in that:
+* metrics will be pushed rather than scraped
+* Prometheus labels will be (ab)used to collect information 
+* the aggregation push gateway will be used to form the metrics into a useful time series
+
+The proposed metrics fall into three categories:
+
+#### Commands
+
+##### cluster_installation_create 
```

Comment:
`cluster_installation_create` does not seem like a great name to me. Any suggestions? 

We could perhaps follow a convention:
`cluster_installation_create_cluster`
`cluster_installation_create_manifests`
`cluster_installation_create_ignition`

Reply:
This is an info metric, IIUC? If yes, then going by the metric naming guidilenaces should be _info suffix e.g. :

> foobar_build_info (for a pseudo-metric that provides metadata about the running binary)

https://prometheus.io/docs/practices/naming/#metric-names

Reply:
The metric is measuring invocation duration, right?  So from "should use base units..." and "should have a suffix describing the unit, in plural form..." sections of those metric-names docs, this should be `installer_create_seconds`, or something, with a `target="cluster"` (or manifests, or ignition-configs, etc.) label.

Reply:
> The metric is measuring invocation duration, right?

Oops, I see that the value is just success (1) or failure (0), so `installer_create_info` or some such makes sense to me.